Polestar: correct charging status

loebse opened the pull request:
Due to https://github.com/evcc-io/evcc/discussions/21504 I discovered that we are not setting the correct charging status C when the car is actually charging.

I also removed the charger status that is no longer available

loebse committed:
correct charging status

loebse commented:
@andig this is an issue that slipped in my recent change:
https://github.com/evcc-io/evcc/discussions/21504
https://github.com/evcc-io/evcc/pull/21413

mfuchs1984 commented:
To make this run, I think you need to add the welcomecharge feature to the template, with default true. Otherwise it will still not be possible to detect the vehicle.

Nevertheless, to review the API implementation properly, a polestar trace log with the vehicle connected but not charging and with vehicle connected and charging would be great.

loebse committed:
welcomecharge

loebse commented:
There is still a lot to learn for me about EVCC 😉 

I hope I am looking at the correct place but I could not find a template of another vehicle that is using the ``welcomecharge`` feature.

I now added the following to the polestar vehicle template: https://github.com/evcc-io/evcc/pull/21517/commits/9a50734b05fab5f5b983eced4ebf389dd253ef9e

Tbh, I am not sure this is the right way to do it an with my wallbox that always must be started with an RFID token, this is hard to test. Can you help @mfuchs1984 ?

Regarding the debug, I added some details to the discussion already.

andig commented:
I don't know. Welcome charge ensures that the vehicle will charge on connect (if not full). If the API responds with status C that should match for detection, but all this needs to happen within 30s. We can sure give it a try...

loebse committed:
Merge branch 'master' into Polestar-charging-status

loebse committed:
Merge branch 'evcc-io:master' into Polestar-charging-status

loebse committed:
Check for empty array and add comments

loebse commented on templates/definition/vehicle/polestar.yaml:
```diff
@@ -7,6 +7,8 @@ params:
   - preset: vehicle-base
   - name: vin
     example: LPSVS...
+  - name: welcomecharge
```
Comment added: 5530683

loebse replied:
@andig is this the comment you expected?

loebse commented on vehicle/polestar/provider.go:
```diff
@@ -36,12 +36,14 @@ var _ api.ChargeState = (*Provider)(nil)
-		res = api.StatusB
-	}
+
+	// Status B currently not possible to read from the API
```
I added a check to make sure the array is not empty: 5530683

However, I don't really understand why we would not need the ``if``

loebse replied:
How should this work, getting rid of the ``if`` @andig

loebse committed:
Merge branch 'master' into Polestar-charging-status
